native/cocos/renderer/gfx-wgpu/WGPUDescriptorSet.cpp

```diff
@@ -65,107 +65,49 @@ void CCWGPUDescriptorSet::clearCache() {
-            dsLayout->updateSamplerLayout(smpEntry.binding, sampler);
         } else if (hasFlag(DESCRIPTOR_BUFFER_TYPE, bindings[i].descriptorType)) {
-            CCWGPUBuffer *buffer = hasFlag(DescriptorType::DYNAMIC_STORAGE_BUFFER | DescriptorType::STORAGE_BUFFER, bindings[i].descriptorType)
-                                       ? deviceObj->defaultResources.storageBuffer
-                                       : deviceObj->defaultResources.uniformBuffer;
-            WGPUBindGroupEntry bufferEntry = {
-                .binding = bindings[i].binding,
-                .buffer = buffer->gpuBufferObject()->wgpuBuffer,
-                .size = buffer->getSize(),
-                .offset = buffer->getOffset(),
-            };
-
             if (hasAnyFlags(bindings[i].descriptorType, DescriptorType::DYNAMIC_STORAGE_BUFFER | DescriptorType::DYNAMIC_UNIFORM_BUFFER)) {
-                _dynamicOffsets.push_back({bindings[i].binding, 0});
+                _dynamicOffsets.emplace(i, bindings[i].binding);
             }
-            _gpuBindGroupObj->bindGroupEntries.push_back(bufferEntry);
-        } else if (bindings[i].descriptorType == DescriptorType::STORAGE_IMAGE) {
-            CCWGPUTexture *texture = deviceObj->defaultResources.storageTexture;
-            WGPUBindGroupEntry texEntry = {
-                .binding = bindings[i].binding,
-                .textureView = texture->gpuTextureObject()->selfView,
-            };
-            _gpuBindGroupObj->bindGroupEntries.push_back(texEntry);
-            dsLayout->updateTextureLayout(texEntry.binding, texture);
-        } else if (bindings[i].descriptorType == DescriptorType::TEXTURE) {
-            CCWGPUTexture *texture = deviceObj->defaultResources.commonTexture;
-            WGPUBindGroupEntry texEntry = {
-                .binding = bindings[i].binding,
-                .textureView = texture->gpuTextureObject()->selfView,
-            };
-            _gpuBindGroupObj->bindGroupEntries.push_back(texEntry);
-            dsLayout->updateTextureLayout(texEntry.binding, texture);
-        } else if (bindings[i].descriptorType == DescriptorType::SAMPLER) {
-            CCWGPUSampler *sampler = deviceObj->defaultResources.unfilterableSampler;
-            WGPUBindGroupEntry smpEntry = {
-                .binding = bindings[i].binding,
-                .sampler = sampler->gpuSampler(),
-            };
-            _gpuBindGroupObj->bindGroupEntries.push_back(smpEntry);
-            dsLayout->updateSamplerLayout(smpEntry.binding, sampler);
-        } else {
-            WGPUBindGroupEntry texEntry = {
-                .binding = bindings[i].binding,
-            };
-            _gpuBindGroupObj->bindGroupEntries.push_back(texEntry);
         }
     }
 
+    _gpuDescriptorObj->gpuDescriptors.resize(bindings.size());
+    _gpuDescriptorObj->descriptorIndices = info.layout->getBindingIndices();
+
+    _bindingSatisfied = [](uint32_t) { return true; };
+
     (void)defaultBindGroup();
```

Comment:
Why add `(void)` here?

Reply:
An unused value emitted, a way to supress warnings